paho/message_ids_test.go

```diff
@@ -29,12 +28,7 @@ func TestMidNoExhaustion(t *testing.T) {
 	})
 	require.NotNil(t, c)
 
-	c.serverInflight = semaphore.NewWeighted(10)
-	c.clientInflight = semaphore.NewWeighted(10)
-	c.stop = make(chan struct{})
-	c.publishPackets = make(chan *packets.Publish)
-	go c.incoming()
-	go c.PingHandler.Start(c.Conn, 30*time.Second)
+	fakeConnect(c)
```

Comment:
I think the tests need to be changed so that we aren't relying on initializing the client within the tests. I'd like to rewrite them so we just set up the test server to respond to `CONNECT` packets and call `Client.connect()` to initialize everything.

Reply:
"we aren't relying on manually initializing the client" - the way this currently works is generally pretty convenient (especially being able to tell the 'broker' exactly what response to send). The V3 client tests rely upon an external broker which works OK but it complicates things (need to start a broker, ensure the relevant port is free etc). 
I have actually been working on a new TEST server that extends the current approach (it's more like a real broker but only supports a single connection). This is required because autopaho is currently not well tested and I'm working on a persistent session option (which will need to be thoroughly tested!).